tests/jupyterhub_api/test_create_session.py

```diff
@@ -0,0 +1,214 @@
+"""
+Create a session
+"""
+
+"""
+To run the test run the following command:
+
+ python3 test_create_session.py --port 443 --users user2 --path /srv/jupyterhub --json '{"LCG-rel": "LCG_95a", "platform": "x86_64-centos7-gcc7-opt", "scriptenv": "none", "ncores": 2, "memory": 8589934592, "spark-cluster": "none"}'
+
+
+For multiple users
+
+python3 test_create_session.py --port 443 --users user0 user1 user2 --path /srv/jupyterhub 
+--token {token value} --json '{"LCG-rel": "LCG_95a", "platform": "x86_64-centos7-gcc7-opt", "scriptenv": "none", "ncores": 2, "memory": 8589934592, "spark-cluster": "none"}'
+"""
+
+
+import json
+import requests
+import subprocess
+import os
+import argparse
+import sys
+sys.path.append("..")
+from logger import Logger, LOG_FOLDER, LOG_EXTENSION
+from SessionUtils import Session
+from requests.packages.urllib3.exceptions import InsecureRequestWarning
+requests.packages.urllib3.disable_warnings(InsecureRequestWarning)
+import json
+import time
+
+
+def get_args():
+    parser = argparse.ArgumentParser(description='Arguments', formatter_class = argparse.ArgumentDefaultsHelpFormatter)
+    parser.add_argument("--port", dest="port", type=int,
+                        required = True,
+                        )
+    parser.add_argument("--path", dest="path",
+                        required=True,
+                        help='path of jupyterhub config file')
+
+    parser.add_argument( "--users", nargs='+', dest="users",
+                        required  = True,
+                        help='list of users')
+    parser.add_argument("--delay", dest="delay",
+                        required=True,
+                        help='delay while creating the server')
+
+    parser.add_argument("--json", dest="json",
+                         required=True,
+                         help = 'json data')
+    args = parser.parse_args()
+    return args
+
+
+class CreateSession:
+    def __init__(self, port, users, path, params, delay, verify):
+        self.port = port
+        self.users = users
+        self.main_url = "https://localhost:" + str(self.port) + "/hub/api/"
+        self.ref_test_name = "Session_creation_test"
+        self.exit = 0
+        self.verify = verify
+        self.path = path
+        self.token = ""
+        self.delay = delay
+        self.data = params
+        self.ref_timestamp = int(time.time())
+        self.session = Session(self.port, users, path, params, verify)
+        self.logger_folder = os.path.join(os.getcwd(), LOG_FOLDER)
+        self.log = Logger(os.path.join(self.logger_folder, self.ref_test_name +"_" + time.strftime("%Y-%m-%d_%H:%M:%S")+ LOG_EXTENSION))
+        self.log_params()
+
+    def log_params(self):
+        self.log.write("parameters", "Test name: " + self.ref_test_name)
+        self.log.write("parameters", "Test time: " + str(self.ref_timestamp))
+        self.log.write("parameters", "Logger folder: " + self.logger_folder)
+
+
+    def create_token(self):
```

Comment:
This should no longer be needed, doesn't it?

Reply:
You should just have one 'load_token' function used by all the tests that need the token

Reply:
Please create a unique function to load the token that is called by all the tests: test_stop_session, test_create_session, test_check_session

Reply:
@ebocchi As per minutes "the tests needing the token will call `jupyterhub token …` to set it up". I could use the same function in all the tests, but create_token function (test_create_session) is creating the token and updating it in the yaml file, the same token is used by the function (load_token), I can create a unique function to load token which can be used by test_create_session and test_check_session.

Reply:
@Divya063
Ok, I think there is a misunderstanding here...
Tell me if I am right:
- we need to write the token together with the admin user name in the 'jupyterhub_config.py' file
- we need to use it as a header for every request that goes to the Hub.

If this is the case, make sure that the testing framework always reads the token from the yaml and never executes `jupyterhub token ...`

Reply:
@ebocchi Second point is correct, first point is wrong, as I mentioned in meeting that if we are executing "jupyterhub token ..", we do not need to modify the jupyterhub_config.py file, it will bind the token with the admin user name and this will automatically get updated the database. I'm clarifying here:
There are two ways to generate token:
1. Use `openssl` command and update the jupyterhub_config.py file with token and corresponding user name, then restart the jupyterhub. In this case it is mandatory to update the jupyterhub_config.py file. 
2. Generate token using the command `jupyterhub token dummy_admin` and use the generated token directly as a header. We can update the jupyterhub_config.py file (if we want) with  token but it is not necessary.